common/src/main/java/io/pravega/common/tracing/RequestTracker.java

```diff
@@ -167,10 +171,14 @@ public void trackRequest(String requestDescriptor, long requestId) {
             List<Long> requestIds = ongoingRequests.getIfPresent(requestDescriptor);
             if (requestIds == null) {
                 requestIds = Collections.synchronizedList(new ArrayList<>());
```

Comment:
As we recently discovered `Collections.synchronizedList(new ArrayList<>());` is not thread safe when iterators are used. 
Please double check if we are using iterators on this collection anywhere. 
either add synchronized while using or change this to Vector something more thread safe.

Reply:
There is no iteration over the list of `requestIds` associated to a request.